src/main/java/com/polishuchenko/bookstore/repository/book/specification/PriceSpecificationProvider.java

```diff
@@ -0,0 +1,39 @@
+package com.polishuchenko.bookstore.repository.book.specification;
+
+import com.polishuchenko.bookstore.model.Book;
+import com.polishuchenko.bookstore.repository.book.SpecificationProvider;
+import java.util.Arrays;
+import java.util.Comparator;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class PriceSpecificationProvider implements SpecificationProvider<Book> {
+    private static final String KEY = "price";
+
+    @Override
+    public Specification<Book> getSpecification(String[] params) {
+        if (params.length <= 1) {
```

Comment:
if params.length == 0, you will call that block too?
```suggestion
        if (params.length == 1) {
```